hypothesis-python/src/hypothesis/strategies/_internal/numbers.py

```diff
@@ -297,7 +305,15 @@ def do_draw(self, data):
             return result
 
     def filter(self, condition):
+        # Handle a few specific weird cases.
+        if condition is math.isfinite:
+            return self
+        if condition in [math.isinf, math.isnan]:
+            return nothing()
+
         kwargs, pred = get_float_predicate_bounds(condition)
+        if pred == UNSATISFIABLE.predicate:
+            return nothing()
```

Comment:
I think we can allow this to be handled via the "min_bound > max_bound" check, no need for a separate case.

Reply:
There are some cases where get_float/int_predicate_bound will return "lambda x: False", like when min_bound is -inf. This check allows those to be passed as empty strategies instead of always-failing strategies. Would that be undesired behavior?

Reply:
I can see a case either way, let's go with empty strategies for consistency - the error messages change a little but not the exception type (if you'd get an error - if not it's a plain perf win).